arch/arm/include/arm/irq.h

```diff
@@ -244,6 +244,12 @@ static inline_function bool up_interrupt_context(void)
 #endif
 }
 
+noinstrument_function
+static inline_function void up_set_interrupt_context(bool flag)
```

Comment:
but why move arch internal function to public header file? let's change to arm_set_interrupt_context and keep in arm_internal.h

Reply:
1. up_set_interrupt_context()/up_interrupt_context() are pair functions. It is easier to read if they are defined in the same file place.
2. up_set_interrupt_context() is implemented differently on each micro-architecture
  a. cortex-a/r uses TPIDRPRW
  b. cortex-m uses ipsr, and this architecture does not need to implement up_set_interrupt_context()
  c. The classic arm architecture and variant architecture tlsr use the legacy global variable g_interrupt_context

The implementation on each variant architecture is different, why is it defined in the common file?